docs/decisions/20231013-authorization-roles.md

```diff
@@ -0,0 +1,165 @@
+---
+status: {proposed}
+date: {2023-10-13}
+---
+
+# Authorization Roles
+
+## Authorization
+
+Until version `v1.10.0` Epinio has a very simple authorization mechanism. A user can be `admin` or `user`.  
+An admin can perform everything, while a user can do anything but only in the namespaces he created or where has permissions.
+
+This was pretty simple to implement but not very flexible.
+
+## Decision Drivers
+
+A new more flexible authorization was needed (https://github.com/epinio/epinio/issues/1946).
+
+Some of the requirements:
+- Be easy to understand and administer
+- Provide least privileges
+- Allow separate roles for listing, reading details, editing, and creating
+- Allow soft multi-tenancy across namespaces (e.g. user A has read write access for apps on one namespace but read only on another namespace)
+- Not require too many changes to the existing code base
+
+## Considered Options
+
+- Secret with whitelist ACL per namespace
+- Role Bindings as their own CRDs
+- ConfigMaps defined Roles with predefined Actions
+
+## Decision Outcome
+
+Chosen option 3 (__"ConfigMaps defined Roles with predefined Actions"__). It was the easier and more flexible solution. It will not require many changes in the code, and it will be easy to customize and implement.
+We could also define some basic roles that can be used.
+
+
+## Pros and Cons of the Options
+
+### Secret with whitelist ACL per namespace
+
+A secret per namespace containing the whitelist ACL could provide this without too much administrative overhead.
+
+The ACL data in the secret could look something like this
+
+```yaml
+application:
+  jwt_claim_name: rwlcd
+  jwt_claim2: l
+service:
+  jwt_claim_name: rw
+  jwt_claim2: rld
+```
+
+Where the RHS of the claim mapping is a string with a combination of letters:
+
+```
+r -- read
+w -- write/edit
+l -- list
+c -- create
+d -- delete
+a -- all
+```
+
+A user with multiple claims would get access to the union of access and an admin would have access to everything.
+
+Additionally, as long as the secret/configmap is mounted in the pod without any subpath, changing it would not require a restart of the API.
+
+#### Pros
+
+* Clear "resource" and "permission" description
+
+#### Cons
+
+* Not sure how to handle newly created namespace
+* Some operations are not strictly tight to a "resource". How to avoid the `exec` but allowing the creation of apps?
+
+### Role Bindings as their own CRDs
+
+Building custom Role Bindings as their own CRDs.
+
+#### Pros
+
+* Clear and structured implementation
+
+#### Cons
+
+* Difficult/complex to implement and evolve
+* Complex to manage
+
+### ConfigMaps defined Roles with predefined Actions
+
```

Comment:
Generally speaking this part needs quite a bit more elaboration of syntax and semantics of the config maps, the action.yaml, the actions (if there is a difference).

Treat this part like writing a specification with __all__ the details.
Currently a lot seems to be missing and glossed over.